standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/properties/HMSServletTest.java

```diff
@@ -92,30 +97,38 @@ protected void stopServer(int port) throws Exception {
     }
   }
 
+  protected static HttpClient createHttpClient(Configuration conf) {
+    SSLContext sslCtxt = clientSSLContextFactory(conf);
+    return sslCtxt != null
+      ? HttpClients.custom().setSSLContext(sslCtxt).setSSLHostnameVerifier(NoopHostnameVerifier.INSTANCE).build()
+      : HttpClients.createDefault();
+  }
 
   @Override
   protected PropertyClient createClient(Configuration conf, int sport) throws Exception {
     String path = MetastoreConf.getVar(conf, MetastoreConf.ConfVars.PROPERTIES_SERVLET_PATH);
-    URL url = new URL("http://hive@localhost:" + sport + "/" + path + "/" + NS);
+    String scheme = MetastoreConf.getBoolVar(conf, MetastoreConf.ConfVars.HTTPSERVER_USE_HTTPS)
+            ? "https" : "http";
+    URI uri = new URI(scheme + "://hive@localhost:" + sport + "/" + path + "/" + NS);
```

Comment:
curious why the variable name is "sport", Sport in english means something totally different :-)